controllers/managedocs_controller.go

```diff
@@ -46,100 +47,155 @@ import (
 )
 
 const (
+	ManagedOCSFinalizer = "managedocs.ocs.openshift.io"
+)
+
+const (
+	managedOCSName         = "managedocs"
 	storageClusterName     = "ocs-storagecluster"
+	prometheusName         = "managed-ocs-prometheus"
+	alertmanagerName       = "managed-ocs-alertmanager"
+	alertmanagerConfigName = "managed-ocs-alertmanager-config"
 	storageClassSizeKey    = "size"
 	deviceSetName          = "default"
 	storageClassRbdName    = "ocs-storagecluster-ceph-rbd"
-	storageClassCephFsName = "ocs-storagecluster-cephfs"
-	deployerCsvPrefix      = "ocs-osd-deployer"
-	ManagedOcsFinalizer    = "managedocs.ocs.openshift.io"
+	storageClassCephFSName = "ocs-storagecluster-cephfs"
+	deployerCSVPrefix      = "ocs-osd-deployer"
+	monLabelKey            = "app"
+	monLabelValue          = "managed-ocs"
 )
 
 // ManagedOCSReconciler reconciles a ManagedOCS object
 type ManagedOCSReconciler struct {
-	client.Client
-	UnrestrictedClient      client.Client
-	Log                     logr.Logger
-	Scheme                  *runtime.Scheme
-	AddonParamSecretName    string
-	DeleteConfigMapName     string
-	DeleteConfigMapLabelKey string
-	AddonSubscriptionName   string
-
-	ctx        context.Context
-	managedOCS *v1.ManagedOCS
-	namespace  string
+	Client             client.Client
+	UnrestrictedClient client.Client
+	Log                logr.Logger
+	Scheme             *runtime.Scheme
+
+	AddonParamSecretName         string
+	AddonConfigMapName           string
+	AddonConfigMapDeleteLabelKey string
+	DeployerSubscriptionName     string
+
+	ctx            context.Context
+	managedOCS     *v1.ManagedOCS
+	storageCluster *ocsv1.StorageCluster
+	prometheus     *promv1.Prometheus
+	alertmanager   *promv1.Alertmanager
+	// alertmanagerConfig *promv1a1.AlertmanagerConfig
+	namespace         string
+	reconcileStrategy v1.ReconcileStrategy
 }
 
 // Add necessary rbac permissions for managedocs finalizer in order to set blockOwnerDeletion.
 // +kubebuilder:rbac:groups=ocs.openshift.io,namespace=system,resources={managedocs,managedocs/finalizers},verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=ocs.openshift.io,namespace=system,resources=managedocs/status,verbs=get;update;patch
 // +kubebuilder:rbac:groups=ocs.openshift.io,namespace=system,resources=storageclusters,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups="monitoring.coreos.com",namespace=system,resources={alertmanagers,prometheuses},verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups="monitoring.coreos.com",namespace=system,resources={podmonitors,servicemonitors,prometheuserules},verbs=get;list;watch;update;patch
```

Comment:
typo, change `prometheuserules` to `prometheusrules`

Reply:
Addressed in https://github.com/openshift/ocs-osd-deployer/pull/30